src/ops/word.rs

```diff
@@ -0,0 +1,64 @@
+pub trait FromWord {
+  fn from_word(bits: u64) -> Self;
+}
+
+pub trait ToWord {
+  fn to_word(self) -> u64;
+}
+
+macro_rules! impl_signed {
+  ( $($ty:ty),+ ) => {
+    $(
+      impl FromWord for $ty {
+        #[inline(always)]
+        fn from_word(bits: u64) -> Self {
+          unsafe { std::mem::transmute::<_, i64>(bits) as Self }
+        }
+      }
+
+      impl ToWord for $ty {
+        #[inline(always)]
+        fn to_word(self) -> u64 {
+          unsafe { std::mem::transmute(self as i64) }
+        }
+      }
+    )*
+  };
+}
+
+macro_rules! impl_unsigned {
+  ( $($ty:ty),+ ) => {
+    $(
+      impl FromWord for $ty {
+        #[inline(always)]
+        fn from_word(bits: u64) -> Self {
+          bits as Self
+        }
+      }
+
+      impl ToWord for $ty {
+        #[inline(always)]
+        fn to_word(self) -> u64 {
+          self as u64
+        }
+      }
+    )*
+  };
+}
+
+impl_signed! { i8, i16, i32 }
+impl_unsigned! { u8, u16, u32, u64 }
```

Comment:
there's no need for a separate macro for signed; a simple `as` cast works for them as well
```suggestion
macro_rules! impl_int {
  ( $($ty:ty),+ ) => {
    $(
      impl FromWord for $ty {
        #[inline(always)]
        fn from_word(bits: u64) -> Self {
          bits as Self
        }
      }

      impl ToWord for $ty {
        #[inline(always)]
        fn to_word(self) -> u64 {
          self as u64
        }
      }
    )*
  };
}

impl_int! { i8, i16, i32, u8, u16, u32, u64 }
```